### ISSUE TITLE
Option to upload data layer

### ISSUE THREAD

**Issue by ClaraIV** [kartoza/GEEST2#375]
For each factor we need to provide the option for layer selection/upload in the “show the properties” box. 
Data Layers might have different names  for different users and so it's important to allow the user to choose which data layer should be used for which analytic.

![Image](https://github.com/user-attachments/assets/a330fcba-4974-4059-bdab-71767fcac60c)

**Comment by timlinux** [comment]
Just to confirm terminology here, by upload you mean 'Select an existing layer on my hard disk'.

After the user selects a layer, if it is a raster layer we would resample and grid align it so it matches the analysis area and bbox. This resampled product would be stored in our working folder so that we can have neatly organised all of the inputs for aggregation etc.

I think in the case of vector layers we have already various mechanisms planned for them to introduce their data e.g. point/line/polygon rasteriser, point buffers, travel distances. etc.

**Comment by ClaraIV** [comment]
yes, sorry if I was not clear; the user should be able to navigate to any folder and load the data layer they want to use for that specific factor; right now it's not clear how the layer selection (loading) happens

**Comment by timlinux** [comment]
This is how I am proposing to approach it @ClaraIV 

https://github.com/kartoza/GEEST2/pull/393

![image](https://github.com/user-attachments/assets/2b63aa15-b940-4b54-bc35-3fc01886b3ec)


For each widget that needs a layer we can let the user choose either a layer loaded in QGIS or a shapefile / tiff as appropriate.

I prefer the approach of loading layers from QGIS as the 'best way' to encourage users to work because it:

* will let us work with all data formats (web services, databases, gpkg etc.)
* will let us filter just the valid layer types (e.g. point layers only)

**Comment by ClaraIV** [comment]
this still needs to be implemented on the start page, for the ADM0 selection. Currently there is no option to browse and select the layer of interest from a folder.
